memote/support/thermodynamics.py

```diff
@@ -0,0 +1,168 @@
+    COMPOUND_MATCHER = CompoundMatcher()
+
+
+def smallest_compound_ID(kegg_ann_list):
+    """
+    Return an ascending list filtered to contain only KEGG compound IDs.
+
+    KEGG IDs may map to compounds, drugs and glycans prefixed respectively
+    with "C", "D", and "G" followed by at least 5 digits.
+
+    Parameters
+    ----------
+    kegg_ann_list: list
+        A list of mixed KEGG IDs.
+
+    """
+    only_cIDs = [x for x in kegg_ann_list if "C" in x]
+    only_cIDs.sort(key=lambda x: int(x.lstrip('C')))
+    return only_cIDs
+
+
+def get_equilibrator_rxn_string(rxn):
+    """
+    Return a reaction string with at least a partial mapping to KEGG IDs.
+
+    First see if there is an unambiguous mapping to a single KEGG compound ID
+    provided with the model. If not, check if there is any KEGG compound ID in
+    a list of mappings. KEGG IDs may map to compounds, drugs and glycans. KEGG
+    compound IDs are sorted so we keep the lowest that is there. If none of
+    this works try mapping to KEGG via the CompoundMatcher by the name of the
+    metabolite. If the metabolite cannot be mapped we simply don't replace it
+    in the original reaction string.
+
+    Parameters
+    ----------
+    rxn: cobra.Reaction
+        The metabolic reaction under investigation.
+
+    """
+    kegg_rxn = rxn.reaction
+    for met in rxn.metabolites:
+        kegg_ann_id = met.annotation.get("kegg.compound")
+        if isinstance(kegg_ann_id, string_types) and "C" in kegg_ann_id:
+            kegg_rxn = kegg_rxn.replace(met.id, kegg_ann_id)
+        elif type(kegg_ann_id) is list and any("C" in s for s in kegg_ann_id):
+            kegg_rxn = kegg_rxn.replace(
+                met.id, smallest_compound_ID(kegg_ann_id)[0]
+            )
+        elif not getattr(met, "name"):
+            continue
+        else:
+            try:
+                df = COMPOUND_MATCHER.match(getattr(met, "name", None))
+                kegg_match_id = df['CID'].iloc[0]
+                kegg_rxn = kegg_rxn.replace(met.id, kegg_match_id)
+            except Exception:
+                pass
+    # COBRApy reaction strings seem to use slightly different arrows which
+    # are not recognized by the eQuilibrator-API
+    return kegg_rxn.replace('-->', '->').replace('<--', '<-')
+
+
+def find_incorrect_thermodynamic_reversibility(reactions, lngamma=3):
+    u"""
+    Return reactions whose reversibilities do not agree with thermodynamics.
+
+    This function checks if the reversibility attribute of each reaction
+    in a list of cobrapy reactions agrees with a thermodynamics-based
+    calculation of the reversibility. To determine reversibility we calculate
+    the reversibility index lngamma (see [1]_ section 3.5) of each reaction
+    using the eQuilibrator API [2]_. The default cutoff for lngamma
+    "corresponds to allowing concentrations to span three orders of magnitude
+    around 100 μM (~3 μM—3mM)" at "pH = 7, I = 0.1M and T = 298K" (see [1]_
```

Comment:
```suggestion
    around 100 μM (~3 μM—3 mM)" at "pH = 7, I = 0.1 M and T = 298 K" (see [1]_
```
SI units should have a space between value and unit.

Reply:
Probably also in above docstring.

Reply:
Yeah definitely also in the docstring. Likewise with the change to `lngamma`